packages/portalnetwork/src/subprotocols/contentLookup.ts

```diff
@@ -79,8 +81,19 @@ export class ContentLookup {
           this.logger(`received uTP connection ID from ${shortId(nearestPeer!.nodeId)}`)
           finished = true
           nearestPeer.hasContent = true
-          return res.value
+          return new Promise((resolve) => {
+            const utpDecoder = (
```

Comment:
Is it an technical choice or style to define this in the return, instead of defining `utpDecoder` on its own?

Maybe i'm just being a n00b here, but this kind of syntax is a little confusing to me.

Reply:
I thought about defining it as a general utility method of sorts but this is a pattern I learned for cleaning up event listeners after their intended purpose is complete. You have to pass the exact function you put on the listener to the `removeListener` or it won't work and we don't want to remove all Content Added listeners. There are probably better ways to do it but I haven't found one yet.